mne/io/nirx/nirx.py

```diff
@@ -299,43 +330,68 @@ def prepend(li, str):
             'sd_index': req_ind,
             'files': files,
             'bounds': bounds,
+            'nan_mask': nan_mask,
         }
+        # Get our saturated mask
+        annot_mask = None
+        for key in ('wl1', 'wl2'):
+            if nan_mask.get(key, None) is None:
+                continue
+            nan_mask[key] = np.isnan(_read_csv_rows_cols(
+                nan_mask[key], 0, last_sample + 1, req_ind, {0: 0, 1: None}).T)
+            if saturated == 'annotate':
+                if annot_mask is None:
+                    annot_mask = nan_mask[key]
+                else:
+                    annot_mask |= nan_mask[key]
+                nan_mask[key] = None  # shouldn't need again
 
         super(RawNIRX, self).__init__(
             info, preload, filenames=[fname], last_samps=[last_sample],
             raw_extras=[raw_extras], verbose=verbose)
 
+        # make onset/duration/description
+        onset, duration, description = list(), list(), list()
+        if annot_mask is not None:
+            on, dur = _mask_to_onsets_offsets(annot_mask.any(0))
+            on = on / info['sfreq']
+            dur = dur / info['sfreq']
+            dur -= on
+            onset.extend(on)
+            duration.extend(dur)
+            description.extend(['BAD_NAN'] * len(on))
```

Comment:
Would this be more specific? Maybe helpful if there are other causes of NaN in other data types M/EEG etc.
```suggestion
            description.extend(['BAD_SATURATION'] * len(on))
```